test/integration/etcd/etcd_cross_group_test.go

```diff
@@ -119,11 +119,14 @@ func TestCrossGroupStorage(t *testing.T) {
 				}
 			}
 
-			versioner := etcd3.APIObjectVersioner{}
+			versioner := storage.APIObjectVersioner{}
 			for _, resource := range resources {
 				// clear out the things cleared in etcd
 				versioned := versionedData[resource.Mapping.Resource]
-				versioner.PrepareObjectForStorage(versioned)
+				if err := versioner.PrepareObjectForStorage(versioned); err != nil {
```

Comment:
Everything else was mechanical refactoring, but I noticed this unhandled error and took care of it as well.

Reply:
great - thanks!